host/app/components/card-catalog-editor.gts

```diff
@@ -0,0 +1,133 @@
+import Component from '@glimmer/component';
+import { getSearchResults } from '../resources/search';
+import type { ExportedCardRef } from '@cardstack/runtime-common';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import CardEditor from './card-editor';
+import ImportModule from './import-module';
+import { LinkTo } from '@ember/routing';
+import { service } from '@ember/service';
+import LocalRealm from '../services/local-realm';
+import type RouterService from '@ember/routing/router-service';
+import { RealmPaths } from '@cardstack/runtime-common/paths';
+import { Loader } from '@cardstack/runtime-common/loader';
+//@ts-ignore cached not available yet in definitely typed
+import { tracked, cached } from '@glimmer/tracking';
+//@ts-ignore glint does not think this is consumed-but it is consumed in the template
+import { hash } from '@ember/helper';
+
+interface Signature {
+  Args: {
+    ref: ExportedCardRef;
+  }
+}
+
+export default class CardCatalogEditor extends Component<Signature> {
+  <template>
+    <ul>
+      {{#each this.catalogEntry.instances as |entry|}}
+        <li>
+          <LinkTo @route="application" @query={{hash path=(this.modulePath (ensureJsonExtension entry.id))}}>
+            {{entry.id}}
+          </LinkTo>
+          <fieldset>
+            <legend>Catalog Entry Editor</legend>
+            <ImportModule @url={{entry.meta.adoptsFrom.module}}>
+              <:ready as |module|>
+                <CardEditor
+                  @card={{hash type="existing" url=entry.id json=(hash data=entry) format="edit"}}
+                  @module={{module}}
+                  @onSave={{this.onSave}}
+                />
+              </:ready>
+              <:error as |error|>
+                <h2>Encountered {{error.type}} error</h2>
+                <pre>{{error.message}}</pre>
+              </:error>
+            </ImportModule>
+          </fieldset>
+          {{!-- TODO: Catalog Entry Preview --}}
```

Comment:
I imagine this is the future refactor to move the preview logic out of the card-ref gts file and move it into here? this would be a good future refactor as well.